homeassistant/components/vizio/media_player.py

```diff
@@ -37,32 +40,47 @@
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 
-SUPPORTED_COMMANDS = SUPPORT_TURN_ON | SUPPORT_TURN_OFF \
-                     | SUPPORT_SELECT_SOURCE \
-                     | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK \
-                     | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP \
-                     | SUPPORT_VOLUME_SET
+SUPPORTED_COMMANDS = {
+    'soundbar': (SUPPORT_PLAY | SUPPORT_PAUSE
+                 | SUPPORT_SELECT_SOURCE
+                 | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+                 | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
+                 | SUPPORT_VOLUME_SET),
+    'tv': (SUPPORT_PLAY | SUPPORT_PAUSE
+           | SUPPORT_SELECT_SOURCE
+           | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+           | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK
+           | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
+           | SUPPORT_VOLUME_SET)
+}
+
+MAX_VOLUME = {
+    'soundbar': 31.0,
+    'tv': 100.0
+}
```

Comment:
IMO these should be defined in the backend library.

Reply:
The reason that these are required is because HA scales the volume slider from 0-100. The base package only supports stepping the volume up or down x times without any knowledge of the max, so the max is solely required for HA.

Reply:
Why I think these belong to the backend library is that this information can be useful for some non-homeassistant user of the library.